assets/animations.js

```diff
@@ -36,7 +38,59 @@ function initializeScrollAnimationTrigger(rootEl = document, isDesignModeEvent =
   animationTriggerElements.forEach((element) => observer.observe(element));
 }
 
-window.addEventListener('DOMContentLoaded', () => initializeScrollAnimationTrigger());
+// Zoom in animation logic
+function initializeScrollZoomAnimationTrigger() {
+  const animationTriggerElements = Array.from(document.getElementsByClassName(SCROLL_ZOOM_IN_TRIGGER_CLASSNAME));
+  const scaleAmount = 0.2 / 100;
+  if (animationTriggerElements.length === 0) return;
+
+  let scrollPositionY = 0;
+
+  animationTriggerElements.forEach((element) => {
+    let elementIsVisible = false;
+    const observer = new IntersectionObserver((elements) => {
+      elements.forEach((entry) => {
+        elementIsVisible = entry.isIntersecting;
+      });
+    });
+    observer.observe(element);
+
+    element.style.setProperty('--zoom-in-ratio', 1 + scaleAmount * percentageSeen(element));
+
+    window.addEventListener('scroll', () => {
+      if (!elementIsVisible) return;
+
+      scrollPositionY = window.scrollY;
+      element.style.setProperty('--zoom-in-ratio', 1 + scaleAmount * percentageSeen(element));
+    });
+  });
+}
+
+function percentageSeen(element) {
+  const viewportHeight = window.innerHeight;
+  const scrollY = window.scrollY;
+  const elementPositionY = element.getBoundingClientRect().top + scrollY;
+  const borderHeight =
+    parseInt(getComputedStyle(element).getPropertyValue('border-top-width')) +
```

Comment:
Might wanna cache `getComputedStyle(element)` in a variable once above and use that variable reference here so we don't need access the computed style twice. Might read a little nicer too.